ner4opt/ner4opt.py

```diff
@@ -0,0 +1,192 @@
+import os
+import pathlib
+from typing import List
+
+import joblib
+
+from .constants import Constants
+from .exceptions import InvalidModelError
+from .features import Featurizer
+from .utils import format_entities, generate_feature_dictionary, l2_augment_sentence, load_torch_model
+
+
+class Ner4Opt(object):
+    """Extract named entities for a given optimization problem.
+
+    The main objective of the Ner4Opt library is to generate named entities for optimization problems.
+    There are six entities the current library extracts; CONST_DIR (constraint direction), LIMIT (limit),
+    OBJ_DIR (objective direction), OBJ_NAME (objective name), PARAM (parameter) and VAR (variable).
+
+    This library provides access to four main models for extracting these entities.
+    `lexical` model built using lexical features alone,
+    `lexical_plus` model which in addition to lexical features uses optimization scenario specific features,
+    `semantic` model which uses a roberta based transformers model and
+    `hybrid` model which combines all the best features of both lexical and semantic models to achieve best results.
+
+    Attributes
+    -------
+    model (str) : Specifies the model to use for extracting the entities.
+                  Options include `lexical`, `lexical_plus`, `semantic` and `hybrid`
+                  Default is `hybrid`
+    use_gpu (bool) : Specifies the model to use gpu while calculating transformer based features.
+                     Default is False
+
+    Methods
+    -------
+    get_entities()
+
+    Attributes
+    -------
+    text (str) : Sample optimization problem
+
+    Returns
+    -------
+
+    Returns named entities of a given optimization problem. A list of dictionaries for each entity
+
+    Keys:
+        Name: start, dtype: int: Starting character index of the entity. Range is [0, len(text)]
+        Name: end, dtype: int: Ending character index of the entity. Range is [0, len(text)]
+        Name: word, dtype: str: Entity phrase
+        Name: entity_group, dtype: str: Type of the entity
```

Comment:
would be good to spell out all 6 entitiy groups here

Reply:
Added the groups